shared/crypto/recipients/index.tsx

```diff
@@ -1,45 +1,58 @@
 import * as React from 'react'
+import * as CryptoGen from '../../actions/crypto-gen'
+import * as Container from '../../util/container'
+import * as Constants from '../../constants/crypto'
 import * as Kb from '../../common-adapters'
 import * as Styles from '../../styles'
-
-type Props = {
-  onAddRecipients: () => void
-  onClearRecipients: () => void
-  recipients: Array<string>
-}
+import {appendEncryptRecipientsBuilder} from '../../actions/typed-routes'
 
 const placeholder = 'Search people'
 
-const Recipients = (props: Props) => (
-  <Kb.Box2 direction="vertical" fullWidth={true}>
-    <Kb.Box2 direction="horizontal" alignItems="center" fullWidth={true} style={styles.recipientsContainer}>
-      <Kb.Text type="BodyTinySemibold" style={styles.toField}>
-        To:
-      </Kb.Text>
-      {props.recipients?.length ? (
-        <Kb.ConnectedUsernames type="BodySemibold" usernames={props.recipients} colorFollowing={true} />
-      ) : (
-        <>
-          <Kb.PlainInput
-            placeholder={placeholder}
-            allowFontScaling={false}
-            onFocus={props.onAddRecipients}
-            style={styles.input}
+const Recipients = () => {
+  const dispatch = Container.useDispatch()
+  // Store
+  const recipients = Container.useSelector(state => state.crypto.encrypt.recipients)
+
+  // Actions
+  const onAddRecipients = React.useCallback(() => {
```

Comment:
just fyi i don't think its necessary to wrap all callbacks in useCallback, only when the perf matters to something thats rendering all the time, and if it really helps. https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
Sure thing